swap/swap.go

```diff
@@ -145,20 +139,43 @@ func New(logpath string, dbPath string, prvkey *ecdsa.PrivateKey, backendURL str
 	if err != nil {
 		return nil, fmt.Errorf("swap init error: %s", err)
 	}
-	if disconnectThreshold <= paymentThreshold {
-		return nil, fmt.Errorf("swap init error: disconnect threshold lower or at payment threshold. DisconnectThreshold: %d, PaymentThreshold: %d", disconnectThreshold, paymentThreshold)
+	if params.DisconnectThreshold <= params.PaymentThreshold {
+		return nil, fmt.Errorf("swap init error: disconnect threshold lower or at payment threshold. DisconnectThreshold: %d, PaymentThreshold: %d", params.DisconnectThreshold, params.PaymentThreshold)
 	}
 	backend, err := ethclient.Dial(backendURL)
 	if err != nil {
 		return nil, fmt.Errorf("swap init error: error connecting to Ethereum API %s: %s", backendURL, err)
 	}
-	return new(
-		logpath,
+
+	if params != nil && params.InitialDepositAmount == 0 {
```

Comment:
I would say that if the user has given params == 0 via flag, there is no need to prompt again and we can just use 0.

Reply:
Thinking about resolving this one, because the default value is 0. however, this will not allow a non-interactive boot up sequence, where we won't deposit in the chequebook. 
We can send, however, 1 `wei` (practically zero).
Not resolving it yet, but if you have read and agree with this consequence @holisticode , you can resolve it.

Reply:
The user won't notice if she puts zero, because it will be ignored and thus the default is taken which is 0.

However I agree this is a bit intransparent and could lead to problems later (for example if we change the default value to not be zero but something else).

For the sake of this PR I suggest we leave it as-is, but we may want to create an issue and discuss what is a good default value, if we need a default value at all, and how much it should be, and maybe as a consequence we need to fix this here.